swagger/swagger-generator/generator-springmvc/src/main/java/org/apache/servicecomb/swagger/generator/springmvc/processor/parameter/SpringmvcDefaultObjectParameterProcessor.java

```diff
@@ -0,0 +1,115 @@
+
+/**
+ * Flatten a object parameter into a set of flatten simple parameters.
+ * <p/>
+ * <em>Nesting object params and generic Object params are NOT supported.</em>
+ * We support query object just for aggregating query params instead of transporting objects in query param.
+ * So we don't support a generic param whose generic type is complex, but a simple generic type param can be supported.
+ */
+public class SpringmvcDefaultObjectParameterProcessor implements DefaultParameterProcessor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SpringmvcDefaultObjectParameterProcessor.class);
+
+  @Override
+  public void process(OperationGenerator operationGenerator, int paramIndex) {
+    Model paramModel = getParamModel(operationGenerator, paramIndex);
+
+    if (null == paramModel) {
+      LOGGER.warn("cannot find param, provider method is [{}], paramIndex = [{}]",
+          operationGenerator.getProviderMethod().getName(), paramIndex);
+      return;
+    }
+
+    LinkedHashMap<String, AbstractSerializableParameter<?>> resultParamMap = getFlattenParams(paramModel);
+
+    addProviderParams(operationGenerator, resultParamMap);
+  }
+
+  private void addProviderParams(OperationGenerator operationGenerator,
+      LinkedHashMap<String, AbstractSerializableParameter<?>> resultParamMap) {
+    resultParamMap.forEach((paramName, param) -> operationGenerator.addProviderParameter(param));
+  }
+
+  private Model getParamModel(OperationGenerator operationGenerator, int paramIndex) {
+    Type paramType = ParamUtils.getGenericParameterType(operationGenerator.getProviderMethod(), paramIndex);
+    Property property = ModelConverters.getInstance().readAsProperty(paramType);
+    // ensure type
+    if (!RefProperty.class.isInstance(property)) {
+      LOGGER.error("Unsupported property type: [{}], paramIndex is [{}]", property.getClass().getName(), paramIndex);
+      return null;
+    }
+
+    Map<String, Model> models = ModelConverters.getInstance().readAll(paramType);
+
+    // find param root
+    RefProperty refProperty = (RefProperty) property;
+    String refTypeName = refProperty.getSimpleRef();
+    Model paramRoot = null;
+    for (Entry<String, Model> entry : models.entrySet()) {
+      if (refTypeName.equals(entry.getKey())) {
+        paramRoot = entry.getValue();
+        break;
+      }
+    }
+    return paramRoot;
+  }
+
+  private LinkedHashMap<String, AbstractSerializableParameter<?>> getFlattenParams(Model paramModel) {
+    LinkedHashMap<String, AbstractSerializableParameter<?>> flattenParamMap = new LinkedHashMap<>();
+    // traversal the properties of current paramModel
+    // create simple parameters, nesting object param is ignored
+    for (Entry<String, Property> propertyEntry : paramModel.getProperties().entrySet()) {
+      if (ParamUtils.isComplexProperty(propertyEntry.getValue())) {
+        LOGGER.info("A nesting complex field is ignored, field name  = [{}]", propertyEntry.getKey());
```

Comment:
if need to be ignore by swagger, better to annotate by "@JsonIgnore"
if we just log and go on, maybe will cause unnotice problem.

Reply:
Fixed, now an error will be thrown to interrupt boot up procedure